codalab/client/json_api_client.py

```diff
@@ -62,13 +63,19 @@ def __init__(self, type_, id_):
         self.id_ = id_
 
 
-class JsonApiClient(RestClient):
+class JsonApiClient(RestClient, BundleClient):
     """
     Simple JSON API client.
     """
-    def __init__(self, address, authenticate):
+    def __init__(self, address, auth_handler, authenticate):
         self._authenticate = authenticate
 
+        # Used by CodaLabManager to look up the associated server address
```

Comment:
This logic is very broken. Please refactor the codaLabManager or whatever so that you don't have to make the JsonApiClient a BundleClient. That makes no sense to me whatsoever. Move the login method if necessary.